usr/lib/soft_stdll/soft_specific.c

```diff
@@ -2751,14 +2751,17 @@ static CK_RV softtok_hmac_final(SIGN_VERIFY_CONTEXT *ctx, CK_BYTE *signature,
         return CKR_MECHANISM_INVALID;
     }
 
-	/* 
-	   Check if signature is NULL to allow multipart signing without segfault 
-	   C_signFinal must be called twice, one time to know the size and allocate the memory, one another to perform the signature
-	*/
-	if (signature == NULL) 
-	{
-		*sig_len = (CK_ULONG)mac_len;
-		return CKR_OK;
+
+if (signature == NULL) 
```

Comment:
Much better now.
You still have a mixture of tabs and spaces here. Please use 4 spaces to indent, not tabs.  See https://github.com/opencryptoki/opencryptoki/blob/master/doc/coding_style.md for details.